### DISCUSSION TITLE
How to plot "Adobe RGB (1998)" color gamut?

### DISCUSSION THREAD

**Discussion by hminle** [colour-science/colour#740] [Q&A] [answered]
Hi,

I cannot draw Adobe RGB 1998 gamut using this command:

```
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(
    colourspaces=["Adobe RGB (1998)"], 
    standalone=False, title=None)
```
![image](https://user-images.githubusercontent.com/16201681/79793840-3e42a000-831f-11ea-9f48-dcbcc5c1da82.png)

For other colour spaces, it's normal, but not this one.
Why is that?

Thanks

**Comment by sobotka** [comment]
Definitely seems like a bug on the string, as the following works:

`colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(colourspaces=colour.models.ADOBE_RGB_1998_COLOURSPACE)`

**Comment by thomasmansencal** [comment, marked as answer]
Hi @hminle,

It is because the mapping is filtered with a Regex pattern: https://github.com/colour-science/colour/blob/b9ebce4be127aea9a282696dc5ac7e162c68c5c4/colour/plotting/common.py#L687

So you would need to escape the parenthesis.

**Comment by thomasmansencal** [comment]
That being said we could improve the docs a bit here!

**Comment by hminle** [comment]
> colour.models.ADOBE_RGB_1998_COLOURSPACE

Thanks, this way works for me
It's quite weird though

```
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(
    colourspaces=['ProPhoto RGB', 'Display P3', colour.models.ADOBE_RGB_1998_COLOURSPACE, 'sRGB'], 
    standalone=False, title=None, colour_cycle_map=cm)
```

**Comment by thomasmansencal** [comment]
Yeah this is supported so that you can both filter existing RGB Colourspaces while passing a custom one you have just built!

I do it here for example with the Venice colourspaces: https://colab.research.google.com/drive/1ZGTij7jT8eZRMPUkyWlv_x5ix5Q5twMB

**Comment by hminle** [comment]
> Yeah this is supported so that you can both filter existing RGB Colourspaces while passing a custom one you have just built!
> 
> I do it here for example with the Venice colourspaces: https://colab.research.google.com/drive/1ZGTij7jT8eZRMPUkyWlv_x5ix5Q5twMB

Thanks,
Just curious. In your Venice experiment, I see its colour gamut goes outside visible range a lot.
So what's the point to make such a colour space like that?

**Comment by thomasmansencal** [comment]
It not that the colourspace is designed to have such primaries in the first place, it is more a direct consequence of fitting the camera data to the *CIE 1931 2 Degree Standard Observer*: if you compute the by-product primaries from the 3x3 matrix mapping the former to the latter, you might get some that goes way beyond the spectral locus. This is actually quite typical from cameras because they are not really colorimetric.

**Comment by saku123hh** [comment]
Hi, I've encountered the same issue. For other colour spaces, it works normally, but not this one. So, I tried using 
`colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(colourspaces=colour.models.ADOBE_RGB_1998_COLOURSPACE)`, but I got the error:
`AttributeError: module 'colour.models' has no attribute 'ADOBE_RGB_1998_COLOURSPACE'`, After some investigation, I found that for my version of the latest colour package, the correct model is `RGB_COLOURSPACE_ADOBE_RGB1998`. Finally, I used the following approach:`plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(
    colourspaces=colour.models.RGB_COLOURSPACE_ADOBE_RGB1998,
    scatter_kwargs={'s': 1, 'c': 'black'}
)` and it worked. I believe that using `colorspaces=["Adobe RGB (1998)"]` is not possible and is likely a bug.".

**Reply by thomasmansencal** [reply]
Hello,

All those variant work for me:

```
import colour

colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(colourspaces=colour.models.RGB_COLOURSPACE_ADOBE_RGB1998)
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(colourspaces=[colour.models.RGB_COLOURSPACE_ADOBE_RGB1998])
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(colourspaces="Adobe RGB (1998)")
colour.plotting.plot_RGB_colourspaces_in_chromaticity_diagram_CIE1931(colourspaces=["Adobe RGB (1998)"])
```

Which version of `colour` are you using?
